text/0000-ember-data-singleton-record-data.md

```diff
@@ -0,0 +1,450 @@
+  id: string | null;
+  clientId: string | null;
+}
+
+interface RecordDataV1 {
+  unloadRecord(): void;
+  isRecordInUse(): boolean;
+  isEmpty(): boolean;
+  isNew(): boolean;
+
+  getAttr(propertyName: string): any;
+  isAttrDirty(propertyName: string): boolean;
+  changedAttributes(): AttributesChanges;
+  hasChangedAttributes(): boolean;
+  rollbackAttributes(): void;
+
+  @deprecated // this RFC deprecates this method entirely
+  getBelongsTo(propertyName: string): { data: LegacyResourceIdentifierObject };
+  @deprecated // this RFC deprecates this method entirely
+  getHasMany(propertyName: string): { data: LegacyResourceIdentifierObject[] };
+
+  willCommit(): void;
+  didCommit(data: any): void;
+
+  @deprecated // this RFC deprecates this method entirely
+  _initRecordCreateOptions(options: object): object;
+  clientDidCreate(): void;
+
+  @deprecated // this RFC deprecates this method entirely and without replacement
+  getResourceIdentifier(): LegacyResourceIdentifierObject;
+
+  // original RecordData RFC specified setBelongsTo
+  // with the 2nd arg being LegacyResourceIdentifierObject
+  // but that was not what the resulting implementation in
+  // ember-data did
+  setDirtyBelongsTo(propertyName: string, value: RecordData | null): void;
+
+  // original RecordData RFC specified setAttribute
+  // but that was not what the resulting implementation in
+  // ember-data did
+  setDirtyAttribute(propertyName: string, value: any): void;
+
+  // original RecordData RFC specified setHasMany
+  // with the 2nd arg being LegacyResourceIdentifierObject[]
+  // but that was not what the resulting implementation in
+  // ember-data did
+  setDirtyHasMany(propertyName: string, value: RecordData[]): void;
+
+  // original RecordData RFC specified addToHasMany
+  // with the 2nd arg being LegacyResourceIdentifierObject[]
+  // but that was not what the resulting implementation in
+  // ember-data did
+  addToHasMany(propertyName: string, values: RecordData[], startIndex?: number): void;
+
+  // original RecordData RFC specified removeFromHasMany
+  // with the 2nd arg being LegacyResourceIdentifierObject[]
+  // but that was not what the resulting implementation in
+  // ember-data did
+  removeFromHasMany(propertyName: string, values: RecordData[]): void;
+
+  @deprecated // this RFC deprecates this method entirely and without replacement
+  removeFromInverseRelationships(isNew: boolean): void;
+}
+```
+
+</td>
+<td>
+
+```ts
+interface RecordDataV2 {
+  unloadRecord(identifier: Identifier);
+  isRecordInUse(identifier: Identifier): boolean;
+  isEmpty(identifier: Identifier): boolean;
+  isNew(identifier: Identifier): boolean;
+
+  getAttr(identifier: Identifier, propertyName: string): any;
+  isAttrDirty(identifier: Identifier, propertyName: string): boolean;
+  changedAttributes(identifier: Identifier): AttributesChanges;
+  hasChangedAttributes(identifier: Identifier): boolean;
+  rollbackAttributes(identifier: Identifier): void;
```

Comment:
If we're changing the API anyway, let's be consistent about whether they're `Attr`s or `Attributes`

Reply:
I cleaned that up elsewhere but agree, made this edit to improve the consistency